sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -87,8 +87,7 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       case "overwrite" => mode(SaveMode.Overwrite)
       case "append" => mode(SaveMode.Append)
       case "ignore" => mode(SaveMode.Ignore)
-      case "error" | "errorifexists" => mode(SaveMode.ErrorIfExists)
-      case "default" => this
+      case "error" | "errorifexists" | "default" => mode(SaveMode.ErrorIfExists)
```

Comment:
I think the `"default"` mode should be mentioned in the list: https://github.com/apache/spark/pull/26094/files#diff-94fbd986b04087223f53697d4b6cab24R77-R81